tools/ImportTool.Shared/SingleFileSpliter.cs

```diff
@@ -0,0 +1,199 @@
+                {
+                    // put all splited files in one hierarchy
+                    string splitedFileName = _sourceBlob.Name.Split('/').Last();
+                    CloudBlockBlob cloudBlockBlob = _sourceBlob.Container.GetBlockBlobReference($"splited/{splitedFileName}_{SplitSizeInBytes}_{index}");
+                    IEnumerable<BlockMeta> blockMetas;
+
+                    length = SplitSizeInBytes;
+
+                    if (stream.Position + length >= totalLength)
+                    {
+                        blockMetas = SplitFileToBlocks(offset, totalLength - offset);
+                        yield return new Tuple<IEnumerable<BlockMeta>, CloudBlockBlob>(blockMetas, cloudBlockBlob);
+                        break;
+                    }
+
+                    stream.Seek(offset + length - 1, SeekOrigin.Begin);
+
+                    // Reads a line. A line is defined as a sequence of characters followed by
+                    // a carriage return ('\r'), a line feed ('\n'), or a carriage return
+                    // immediately followed by a line feed.
+                    int count = 0;
+                    while (true)
+                    {
+                        int c = stream.ReadByte();
+
+                        // \n - UNIX   \r\n - DOS   \r - Mac
+                        if (c == -1 || (c == 10 || c == 13))
+                        {
+                            if (c == 13 && stream.ReadByte() == 10)
+                            {
+                                count++;
+                            }
+
+                            break;
+                        }
+
+                        count++;
+                    }
+
+                    length += count;
+                    blockMetas = SplitFileToBlocks(offset, length);
+                    offset += length;
+                    index++;
+
+                    yield return new Tuple<IEnumerable<BlockMeta>, CloudBlockBlob>(blockMetas, cloudBlockBlob);
+                }
+            }
+        }
+
+        private IEnumerable<BlockMeta> SplitFileToBlocks(long offset, long length)
+        {
+            List<BlockMeta> blockMetas = new List<BlockMeta>();
+            while (length > BlockSizeInBytes)
+            {
+                blockMetas.Add(new BlockMeta
+                {
+                    Id = Base64UrlEncoder.Encode(Guid.NewGuid().ToString()),
+                    Offset = offset,
+                    Length = BlockSizeInBytes,
+                });
+
+                length -= BlockSizeInBytes;
+                offset += BlockSizeInBytes;
+            }
+
+            blockMetas.Add(new BlockMeta
+            {
+                Id = Base64UrlEncoder.Encode(Guid.NewGuid().ToString()),
+                Offset = offset,
+                Length = length,
+            });
+
+            return blockMetas;
+        }
+
+        private async Task PutNewSplitedFile(IEnumerable<BlockMeta> blockMetas, CloudBlockBlob cloudBlockBlob)
```

Comment:
UploadBlock

Reply:
fidxed, use UploadBlob, it is a blob composed of several blocks